prov/efa/src/efa_prov_info.c

```diff
@@ -455,13 +434,25 @@ int efa_prov_info_alloc(struct fi_info **prov_info_ptr,
 	if (!prov_info)
 		return -FI_ENOMEM;
 
-	if (ep_type != FI_EP_RDM && ep_type != FI_EP_DGRAM) {
-		EFA_WARN(FI_LOG_DOMAIN, "Unsupported endpoint type: %d\n",
-			 ep_type);
-		return -FI_ENODATA;
+	prov_info->mode |= FI_CONTEXT2; 	/* EFA direct path requires FI_CONTEXT2 mode */
```

Comment:
I am not sure whether we should assign FI_CONTEXT2 here as it is not a device attribute. But I don't have a strong opinion as long as we removed it for rdm

Reply:
For safety, I think we need to add this to tx/rx_attr->mode as well.